### PR TITLE
Fix query id extraction

### PR DESCRIPTION
**Pull request by JamesRTaylor**

### CODE REVIEW

File: gateway-ha/src/main/java/com/lyft/data/gateway/ha/handler/QueryIdCachingProxyHandler.java [hunk 1]
```diff
@@ -182,6 +188,10 @@ protected String extractQueryIdIfPresent(HttpServletRequest request) {
           queryId = tokens[3];
         }
       }
+    } else if (path.startsWith(UI_API_QUERY)
+        && path.length() >= UI_API_QUERY.length() + QUERY_ID_LENGTH) {
```
Comment (amalakar) [inline gateway-ha/src/main/java/com/lyft/data/gateway/ha/handler/QueryIdCachingProxyHandler.java:192]: @JamesRTaylor wondering if we want to merge this with the `path.startsWith(PRESTO_UI_PATH)` block. Also presto query id seems to have been pretty stable. Would a regex extraction for anywhere in the path for a tighter regex `20200416_160256_03078_6b4yt` be helpful here, something like: `.*/(\d+_\d+_\d+_\w+).*`

Comment (hitejinder) [reply gateway-ha/src/main/java/com/lyft/data/gateway/ha/handler/QueryIdCachingProxyHandler.java:192]: Yeah, that's another way of doing it.

**Pull request merged**